libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -625,9 +626,12 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				*arguments.front()->annotation().type,
 				*function.parameterTypes().front(), true
 			);
-			// gas <- gas * !value
-			m_context << Instruction::SWAP1 << Instruction::DUP2;
-			m_context << Instruction::ISZERO << Instruction::MUL << Instruction::SWAP1;
+			if (function.location() != Location::Transfer)
```

Comment:
What is the point of this code though? It says provide call stipend (2300) and remove it if we're not sending 0 ether.

In `appendExternalfunctionCall` we'd set it to 9000 should we not set a gas there.